tests/lifecycle/tests/lifecycle_affinity_required_pods.go

```diff
@@ -31,6 +32,10 @@ var _ = Describe("lifecycle-affinity-required-pods", func() {
 		By("Clean namespace before each test")
 		err = namespaces.Clean(tsparams.LifecycleNamespace, globalhelper.GetAPIClient())
 		Expect(err).ToNot(HaveOccurred())
+
+		By("Ensure all nodes are labeled with 'worker-cnf' label")
```

Comment:
optional(not sure that this would follow current code pattern): it seems that we are missing a "BeforeEach" and "AfterEach" at the program level before each test. Most of these are the same for every suite. For each suite, we could rename the namespace param to just "Namespace" from "(tsparams.LifecycleNamespace" to "tsparams.Namespace" . The import path makes this unique. Then implement functions globalHelper.BeforeEach(tsparams.Namespace) and globalHelper.AfterEach(tsparams.Namespace) with everything we need to do globally in every "BeforeEach" and "AfterEach". If some suite specific behavior, then add an "if" statement on the namespace inside these 2 new functions.

Reply:
This could probably be experimented with in another branch/PR.  However on the flip side I sort of like that each test is explicit with what it needs for pre-requisites for bring-up and tear-down.